core/src/Kokkos_CopyViews.hpp

```diff
@@ -1350,14 +1350,13 @@ inline void contiguous_fill(
 }
 
 // Default implementation for execution spaces that don't provide a definition
-template <typename ExecutionSpace, class ViewType>
+template <typename ExecutionSpace>
 struct ZeroMemset {
-  ZeroMemset(const ExecutionSpace& exec_space, const ViewType& dst) {
-    using ValueType = typename ViewType::value_type;
-    alignas(alignof(ValueType)) unsigned char
-        zero_initialized_storage[sizeof(ValueType)] = {};
-    contiguous_fill(exec_space, dst,
-                    *reinterpret_cast<ValueType*>(zero_initialized_storage));
+  ZeroMemset(const ExecutionSpace& exec_space, void* dst, size_t cnt) {
+    contiguous_fill(exec_space,
+                    Kokkos::View<std::byte*, Kokkos::MemoryUnmanaged>(
```

Comment:
Doesn't this need to be: 
```suggestion
                    Kokkos::View<std::byte*, ExecutionSpace, Kokkos::MemoryUnmanaged>(
```